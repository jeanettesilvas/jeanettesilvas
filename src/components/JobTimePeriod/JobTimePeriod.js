import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import './JobTimePeriod.css';

export class JobTimePeriod extends Component {
    static propTypes = {
        dateRange: PropTypes.string.isRequired,
        highlightedText: PropTypes.string,
        summary: PropTypes.string.isRequired,
        tags: PropTypes.arrayOf(PropTypes.string).isRequired,
        teamName: PropTypes.string.isRequired,
    };

    static defaultProps = {
        highlightedText: '',
    };

    get tags() {
        let { tags } = this.props;
        let tagListItems;
        let highlightedText = this.props.highlightedText;
        let className;

        tags.sort();
        tagListItems = tags.map((tag, index) => {
            className = 'job-time-period__tag';

            if (highlightedText) {
                className = classnames(className, {
                    'job-time-period__highlighted-tag': tag.includes(
                        highlightedText
                    ),
                });
            }

            return (
                <li className={className} key={`tag-${index}`}>
                    {tag}
                </li>
            );
        });

        return tagListItems;
    }

    render() {
        let { dateRange, summary, teamName } = this.props;

        return (
            <section className="job-time-period">
                <h3 className="job-time-period__date-range">{dateRange}</h3>
                <h1 className="job-time-period__team">{teamName}</h1>
                <p className="job-time-period__description">{summary}</p>

                <ul className="job-time-period__tags">{this.tags}</ul>
            </section>
        );
    }
}
