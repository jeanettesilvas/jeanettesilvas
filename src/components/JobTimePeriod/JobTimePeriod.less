@import '../../less/_variables';

.job-time-period {
    color: @light-gray-text;
    font-weight: 200;
    margin: @spacing-3x 0 0;
    width: 200px;

    &__date-range {
        font-size: 12px;
        font-weight: 200;
        margin: 0 0 4px;
    }

    &__team {
        color: @white;
        font-weight: 200;
        margin: 0 0 @spacing-2x;
    }

    &__description {
        margin-bottom: @spacing-3x;
    }

    &__tags-title {
        color: @white;
        margin-bottom: @spacing-1x;
    }

    &__tags {
        border-top: 1px solid @gray;
        display: flex;
        flex-wrap: wrap;
        list-style: none;
        margin: 0;
        padding: @spacing-4x 0 0;
    }

    @media screen and (max-width: 729px) {
        width: auto;
    }

    @media screen and (max-width: 990px) and (min-width: 860px) {
        width: 350px;
    }

    @media screen and (max-width: 859px) and (min-width: 730px) {
        width: 300px;
    }
}
