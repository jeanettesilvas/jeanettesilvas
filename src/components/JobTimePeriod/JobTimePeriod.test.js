import React from 'react';
import { shallow } from 'enzyme';
import { JobTimePeriod } from './JobTimePeriod';
import renderer from 'react-test-renderer';

describe('<JobTimePeriod />', () => {
    let props;

    beforeEach(() => {
        props = {
            dateRange: '2018',
            highlightedText: 'a',
            summary: 'summary text',
            tags: ['react', 'foo'],
            teamName: 'team',
        };
    });

    describe('#render', () => {
        it('renders correctly', () => {
            const tree = renderer.create(<JobTimePeriod {...props} />).toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('renders without crashing', () => {
            const wrapper = shallow(<JobTimePeriod {...props} />);
            expect(wrapper.find('.job-time-period').length).toEqual(1);
        });
    });
});
