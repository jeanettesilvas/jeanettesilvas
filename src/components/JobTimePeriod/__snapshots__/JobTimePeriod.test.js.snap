// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobTimePeriod /> #render renders correctly 1`] = `
<section
  className="job-time-period"
>
  <h3
    className="job-time-period__date-range"
  >
    2018
  </h3>
  <h1
    className="job-time-period__team"
  >
    team
  </h1>
  <p
    className="job-time-period__description"
  >
    summary text
  </p>
  <ul
    className="job-time-period__tags"
  >
    <li
      className="tag tag-undefined"
    >
      foo
    </li>
    <li
      className="tag tag-undefined tag--highlighted"
    >
      react
    </li>
  </ul>
</section>
`;
